### ISSUE TITLE
[Adapters] DataLoaderV2 Adapters Ideas

### ISSUE THREAD

**Issue by VitalyFedyunin** [pytorch/data#532]
### 🚀 The feature

[ ] Add `Collate(fn)` Adapter which will add `.collate(fn)` DataPipe at the end of the graph. Optional: we can think about overriding existing collate in the graph for new function.

### Motivation, pitch

As we are growing product there are more and more adapters ideas and requests coming up. This issue created to aggregate such requests and keep track of duplications.

**Comment by NivekT** [comment]
Are we considering `PinMemory` and `Shuffle` (to turn on and off) as adapters as well? Or those will be strictly DataPipes?

**Comment by VitalyFedyunin** [comment]
> Are we considering `PinMemory` and `Shuffle` (to turn on and off) as adapters as well? Or those will be strictly DataPipes?

They should be both adapters and pipes.
